specs/experimental/contracts/safe/liveness-module.md

```diff
@@ -0,0 +1,284 @@
+
+## Overview
+
+The LivenessModule enables permissionless removal of inactive Safe owners based on liveness tracking data from the
+LivenessGuard. When the owner count falls below the minimum threshold, the module transfers sole ownership to a
+designated fallback owner and deactivates itself.
+
+## Definitions
+
+### Liveness Interval
+
+The time period (in seconds) during which an owner must demonstrate activity to avoid being considered inactive. An
+owner is eligible for removal if their last recorded activity timestamp plus the [Liveness Interval] is less than the
+current block timestamp.
+
+### Minimum Owners
+
+The minimum number of owners that must be maintained in the Safe before triggering [Shutdown]. If owner removals would
+reduce the count below this threshold, all remaining owners must be removed and ownership transferred to the [Fallback
+Owner].
+
+### Threshold Percentage
+
+The percentage (1-100) used to calculate the Safe's signing threshold based on the current number of owners. The
+threshold is computed as the smallest integer greater than or equal to `(numOwners * ThresholdPercentage) / 100`.
+
+### Fallback Owner
+
+The address that receives sole ownership of the Safe during [Shutdown]. This address is expected to be controlled by
+governance and serves as the recovery mechanism when the owner set becomes too small.
+
+### Shutdown
+
+The process triggered when owner removals would reduce the owner count below [Minimum Owners]. During shutdown, all
+remaining owners are removed, the [Fallback Owner] becomes the sole owner with a threshold of 1, and the module
+permanently deactivates itself by setting `ownershipTransferredToFallback` to true.
+
+## Assumptions
+
+### a01-001: LivenessGuard provides accurate liveness data
+
+The LivenessGuard contract correctly tracks and reports the last activity timestamp for each Safe owner via its
+`lastLive` mapping. Inaccurate liveness data would cause the module to incorrectly identify owners as inactive or
+active.
+
+#### Mitigations
+
+- The LivenessGuard specification defines invariants ensuring liveness tracking accuracy
+- The module verifies the LivenessGuard has not been changed during the removal process
+- The LivenessGuard address is immutable in the module, preventing substitution
+
+### a02-002: Fallback owner is governance-controlled
+
+The [Fallback Owner] address is controlled by legitimate governance and will act in the best interest of the Safe's
+stakeholders. A malicious fallback owner could gain complete control of the Safe during [Shutdown].
+
+#### Mitigations
+
+- The fallback owner is set at deployment and cannot be changed without deploying a new module
+- Shutdown only occurs when the owner count falls below [Minimum Owners], which requires multiple owner removals
+- The Safe's existing owners can remove the module before shutdown occurs if they maintain sufficient liveness
+
+## Invariants
```

Comment:
devinai 

This one is complex, because the contract hides a lot of higher level things. I'd say the better list of invariants is along the lines of:

- Threshold is maintained
- Owners that aren't live are correctly removed
- Owners that are live are not removed
- Fallback is properly triggered if owner count is unsafe